### ISSUE TITLE
Port mappings: add support of port ranges

### ISSUE THREAD

**Issue by tuusberg** [aws/amazon-ecs-agent#269]
Hi there!

Some applications do require wide range of open ports and currently the only option we have is to set them explicitly (e.g. one by one)  in task definition. 

I think that it would be useful to allow use of port ranges, like
```
{
          "hostPorts": 3000-3025,
          "containerPorts": 3000-3025,
          "protocol": "tcp"
}
```
Also, I've noticed that ecs-agent throws an error if there is > 100 ports specified in task definition, regardless of port numbers: 
```
 Run tasks failed
 Reasons : RESOURCE:PORTS
```

**Comment by euank** [comment]
You're correct that we don't currently support specifying port ranges directly in task definitions.
I made sure the feature of port ranges is being tracked internally.

There is indeed a limit on the number of reserved and utilized ports. This limit, per [the docs](http://docs.aws.amazon.com/AmazonECS/latest/APIReference/API_PortMapping.html), is 50.

If you have more information on your use-case, it could help us in evaluating that limit.

Best,
Euan

**Comment by katiyar-g** [comment]
It would be nice to have this feature as I have a similar use case where the application requires a large number of ports (1 thousand ports) in order to communicate with the another instance running on a different host.

**Comment by margusbirk** [comment]
+1

**Comment by z0r1k** [comment]
Since docker supports port ranges I would expect that it should be possible to specify port ranges by ECS agent too.
Also I can easily image use-case where wide range or ports should be used, for example video conference SFU which can handle up to 10k WebRTC connections.

**Comment by wei-m-teh** [comment]
+1

**Comment by chenliu0831** [comment]
👍

**Comment by dolgafar** [comment]
+1 
WebRTC / RTP
(Should be easy to do, since it's now supported in docker)

**Comment by bhendo** [comment]
+1
Is there any status on the progress of this?
I'll add FTP to the mix of use cases.

**Comment by sebalas** [comment]
+1

**Comment by tobiasc** [comment]
+1

**Comment by RyanBowlby-Reflektion** [comment]
+1

**Comment by jamiegs** [comment]
I would also like this to allow us to restrict containers to different ports so I can define security groups and only allow upstream services to access specific services running in ECS.

**Comment by adamdunkley** [comment]
+1

@euank I know this has been added to the use cases already, but FTP is what we require this for. We're adding passive ports in manually and we can only allow 49 concurrent PASV connections with the 50 port limit (with 1 for the control port on 21).

**Comment by pirasakat** [comment]
+1

**Comment by mihbor** [comment]
+1
At the very least the docs should mention ports parameter syntax is only partially supported.
http://docs.aws.amazon.com/AmazonECS/latest/developerguide/cmd-ecs-cli-compose.html
The way it's written currently implies that the listed parameters are supported as per the linked docker compose syntax (v1&2) docs, which prominently show examples of the use of ranges: https://docs.docker.com/compose/compose-file/compose-file-v2/#ports

**Comment by Dominik-K** [comment]
At least, the 50 ports limit should throw an useful error message in the agent. Currently it just shows this misleading events message (in the AWS console):

> service [...] was unable to place a task because no container instance met all of its requirements. The closest matching container-instance [...] is already using a port required by your task.

**Comment by marccardinal** [comment]
+1

**Comment by faizan82** [comment]
Couchbase server cluster using ECS Weave but managing ports is becoming a pain:  
21100 to 21299 (inclusive) Node to Node communication

**Comment by bs-thomas** [comment]
+1  I'm hosting a ftp proxy daemon, and it needs to support a range of ports for passive mode.  The range is 200 ports, and is impossible to achieve at the moment.  Are there any plans to support this?

**Comment by minzhang28** [comment]
+1

**Comment by aabrahamian** [comment]
+1

**Comment by nswarnkar** [comment]
This is important requirement and we are going to hugely use ECS. Are you guys working on it ? When this could be available?

**Comment by tuusberg** [comment]
I created this issue more than a year ago. Is there a chance that you guys would implement this feature in the near future?

**Comment by brainframe-me** [comment]
For those who are trying to setup an FTP server inside a ECS task container and are trying to allow PASV connections and bump into the limitations of ECS-agent. Please know that there is a simple solution that does not require you to define all potential dynamic pasv ports that you defined in your server configuration. 
You simply need to set the NetworkMode of your TaskDefinition to 'host' instead of 'bridge'. This gives the container access to ports it dynamically opens (as is the case with passive FTP connections).
Obviously in your security group you still need to allow the dynamic range you use, but here you can use a range.
Hope this helps somebody, I lost quite some time trying to get this to work

**Comment by houqp** [comment]
This is also a major blocker for us, is there a timeline for this?

**Comment by phil-lavin** [comment]
Confirmed that using `NetworkMode`: `host` instead of `bridge` does not require you to explicitly expose ports. Rather, all traffic to the host is directly mapped to the same port on the container. We've successfully run an RTP proxy for WebRTC inside ECS.

**Comment by Anamican** [comment]
> For those who are trying to setup an FTP server inside a ECS task container and are trying to allow PASV connections and bump into the limitations of ECS-agent. Please know that there is a simple solution that does not require you to define all potential dynamic pasv ports that you defined in your server configuration.
> You simply need to set the NetworkMode of your TaskDefinition to 'host' instead of 'bridge'. This gives the container access to ports it dynamically opens (as is the case with passive FTP connections).
> Obviously in your security group you still need to allow the dynamic range you use, but here you can use a range.
> Hope this helps somebody, I lost quite some time trying to get this to work

Thank you very much, after struggling for two days, this finally helped !
